src/mon/OSDMonitor.cc

```diff
@@ -2038,22 +2037,6 @@ bool OSDMonitor::should_propose(double& delay)
     return true;
   }
 
-  // propose as fast as possible if updating up_thru or pg_temp
-  // want to merge OSDMap changes as much as possible
-  if ((pending_inc.new_primary_temp.size() == 1
-      || pending_inc.new_up_thru.size() == 1)
-      && pending_inc.new_state.size() < 2) {
-    dout(15) << " propose as fast as possible for up_thru/pg_temp" << dendl;
-
-    utime_t now = ceph_clock_now();
-    if (now - last_attempted_minwait_time > g_conf->paxos_propose_interval
```

Comment:
If there's an issue with the logic, I think it's here — perhaps we need to take the most recent of last_attempted_minwait_time and paxos->get_last_commit_time() in this first check of the if-block?
My head's a little fuzzy right now.

Reply:
yes exactly it correctly letting us do one fast .the problem is how to define `first` commit, the code logic now make it `first` whenever after a regular commit, so it goes to fast/slow/fast/slow loop instead of fast/slow/slow/slow.

And yes we can fix the logical with adding more `if` and make it more fuzzy. but it looks to me like the paxoservice::should_propose already implement what we want and clear enough.
```
 utime_t now = ceph_clock_now();
    if ((now - paxos->last_commit_time) > g_conf->paxos_propose_interval)
      delay = (double)g_conf->paxos_min_wait;
    else
      delay = (double)(g_conf->paxos_propose_interval + paxos->last_commit_time
		       - now);
```

Reply:
Oh I get it now.